spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -128,8 +128,15 @@ public void setBatchingSettings(BatchingSettings batchingSettings) {
 	public Publisher createPublisher(String topic) {
 		return this.publishers.computeIfAbsent(topic, (key) -> {
 			try {
+				String finalProject = this.projectId;
+				String finalTopic = topic;
+				if (topic.contains("/")) {
```

Comment:
Why `split("/")` doesn't work instead?
There is going to be a single element in the result array if there is no `/`.

Reply:
I thought of using `split('/")`, but what if there is more than one `/` used in the topic.

Reply:
There is an overloaded `split()` method then with `limit`:
```
 * <p> The {@code limit} parameter controls the number of times the
     * pattern is applied and therefore affects the length of the resulting
     * array.  If the limit <i>n</i> is greater than zero then the pattern
     * will be applied at most <i>n</i>&nbsp;-&nbsp;1 times, the array's
     * length will be no greater than <i>n</i>, and the array's last entry
     * will contain all input beyond the last matched delimiter.  If <i>n</i>
     * is non-positive then the pattern will be applied as many times as
     * possible and the array can have any length.  If <i>n</i> is zero then
     * the pattern will be applied as many times as possible, the array can
     * have any length, and trailing empty strings will be discarded.
```

Reply:
Thanks, I didn't know about that.